### PR TITLE
Keyboard noises for computers, modular consoles and laptops

### PR DESCRIPTION
**Pull request by Azarak**
## Changelog
:cl:
add: Keyboard noises for computers, modular consoles and laptops
/:cl:

### CODE REVIEW

**Comment by atakiya** [comment]
Does this make a random noise whenever you open the interface only?

**Comment by Azarak** [comment]
> 
> 
> Does this make a random noise whenever you open the interface only?

Any time you open or interact with it

**Comment by atakiya** [comment]
I see, we'd have to check with some interfaces if there's overlap, but this should be fine to merge for now.
One I can think of is the HoP console, which has keyboard noises when you fill out ID information, on typing into the boxes.
It's quite neat.

**Comment by Azarak** [comment]
> 
> 
> I see, we'd have to check with some interfaces if there's overlap, but this should be fine to merge for now.
> One I can think of is the HoP console, which has keyboard noises when you fill out ID information, on typing into the boxes.
> It's quite neat.

That's a fair point

File: code/modules/modular_computers/computers/item/computer.dm [hunk 1]
```diff
@@ -52,6 +52,11 @@
 	var/comp_light_luminosity = 3 //The brightness of that light
 	var/comp_light_color //The color of that light
 
+	/// Keeping track of the cooldown for the keyboard click sound
+	var/next_click = 0
+	/// Whether this modular computer makes click noises.
+	var/makes_click_noises = TRUE
```
Comment (atakiya) [inline code/modules/modular_computers/computers/item/computer.dm:58]: Unused variable? (at least in this PR)

Comment (Azarak) [reply code/modules/modular_computers/computers/item/computer.dm:58]: It's used in the tablet subtype. I assume it uses a touchscreen, so no keyboard noises

Comment (atakiya) [reply code/modules/modular_computers/computers/item/computer.dm:58]: Yeah but I don't see a proc where that is used in?

Comment (atakiya) [reply code/modules/modular_computers/computers/item/computer.dm:58]: image
Only two defines, no checks or anything.

File: code/modules/modular_computers/computers/item/computer_ui.dm [hunk 2]
```diff
@@ -23,6 +23,10 @@
 			ui.close()
 		return
 
+	if(world.time > next_click && in_range(get_turf(src), user))
```
Comment (Azarak) [inline code/modules/modular_computers/computers/item/computer_ui.dm:26]:
```suggestion
	if(makes_click_noises && world.time > next_click && in_range(get_turf(src), user))
```

File: code/modules/modular_computers/computers/item/computer.dm (hunk 1 above)
Comment (Azarak) [reply code/modules/modular_computers/computers/item/computer.dm:58]: You are totally right, fixed now

**Review by atakiya — APPROVED** [review]

**Pull request merged**